packages/typir/src/kinds/function-kind.ts

```diff
@@ -246,13 +248,12 @@ export type InferFunctionCall<T = unknown> = {
  * - optional parameters
  * - parameters which are used for output AND input
  */
-export class FunctionKind implements Kind {
+export class FunctionKind implements Kind, TypeGraphListener {
```

Comment:
In the implementations of this class, the return type is widely disregarded. For example in calculateFunctionKey etc. This is on purpose, right? The return time is not to be the distinguishing factor?

Reply:
For overloaded functions, usually the output/return type of a function does not matter, yes.

If it matters, you should override `calculateFunctionKey` to include the return type in the string as well.

> In the implementations of this class
Do you mean UniqueFunctionValidation or FunctionKind?